api/views/newsletter.py

```diff
@@ -33,20 +34,7 @@ def post(self, request):
             create_contact.update_enabled = True
             api_instance.create_contact(create_contact)
             return EmptyValidResponse
-
         except sib_api_v3_sdk.rest.ApiException as e:
-            contact_exists = json.loads(e.body).get("message") == "Contact already exist"
-            if contact_exists:
-                logger.info(f"Newsletter contact already exists: {email}")
-                return EmptyValidResponse
-            logger.exception("SIB API error in newsletter subsription :\n{e}")
-            return JsonResponse(
-                {"error": "Error calling SendInBlue API"},
-                status=status.HTTP_400_BAD_REQUEST,
-            )
-        except ValidationError as e:
-            logger.warning(f"Invalid email on newsletter subscription: {email}:\n{e}")
-            return JsonResponse({"error": "Invalid email"}, status=status.HTTP_400_BAD_REQUEST)
-        except Exception as e:
-            logger.exception(f"Error on newsletter subscription:\n{e}")
-            return UnknownErrorResponse
+            if json.loads(e.body).get("message") == "Contact already exist":
+                raise EmailAlreadyExists
+            raise ProjectAPIException(global_error="Une erreur inconnue avec l'API SendInBlue a eu lieu.")
```

Comment:
Je ne mettrai pas l'info sur notre fournisseur SMTP sur la réponse user, c'est trop technique. Peut-être quelque chose plus générique comme 
```suggestion
            raise ProjectAPIException(global_error="Une erreur inconnue a eu lieu lors de votre inscription à la newsletter, veuillez réessayer plus tard")
```

Reply:
Ok d'accord avec toi, mais alors dans ce cas si c'est "une erreur inconnue" je suppose c'est pas nécessaire de rajouter "lors de votre inscription à la newsletter" (parce que la personne le sait) et rester sur l'erreur serveur de base sans en créer une nouvelle.